coffee/plan.py

```diff
@@ -93,7 +93,11 @@ def plan_cpu(self, opts):
                 parent, nest = expr_info
                 if not nest:
                     continue
-                metaexpr = MetaExpr(check_type(stmt, info['decls']), parent, nest)
+                if kernel.template:
+                    typ = "double"
```

Comment:
This works "for now", but note going forward this should be parameterised by the scalar type of the output argument of the kernel I think.